pandas/core/arrays/string_arrow.py

```diff
@@ -132,13 +132,9 @@ class ArrowStringArray(
     """
 
     def __init__(self, values) -> None:
+        super().__init__(values)
+        # TODO: Migrate to ArrowDtype instead
```

Comment:
L137 is not now necessary right? (also comment is not useful here)

Reply:
This will still need to be changed in a follow up since `StringDtype` handles python + pyarrow types together and can dispatch to ArrowStringArray.

So there will be an (experimental) breaking change as we want this to eventually go through ArrowDtype dispatching to `ArrowArray`